### PR TITLE
Configure app-aware Vercel deployments

### LINKED ISSUE #239: Show seven peer apps in visual review
Parent: #238

## Dependency
Update open pull request #233 before merge.

## Branch
feature/vercel-app-deployments

## Owned areas
Visual manifest, review rail, route fixtures, and visual-review tests.

## Tasks
- [x] Add appId, appLabel, and captureKind to every route.
- [x] Render one group for each real app.
- [x] Capture Optimitron with its explicit site variant.
- [x] Move old host checks into a compatibility group.
- [x] Preserve filters, links, verdicts, and mobile behavior.

## Acceptance
- [x] The rail displays seven app groups.
- [x] Optimitron has its own visible group.
- [x] Compatibility captures never count as apps.
- [x] Every route links to the correct preview.
- [x] Pull request #233 remains mergeable.

## Delivery
- [x] Assign an owner.
- [x] Mark implementation started.
- [x] Complete focused tests.
- [x] Capture desktop and mobile screenshots.
- [x] Inspect the visual review.
- [x] Link the pull request: #233.
- [x] Verify Vercel previews.
- [ ] Merge after required checks pass.

## Evidence
- Pull request: https://github.com/mikepsinn/optimitron/pull/233
- Published review: https://mikepsinn.github.io/optimitron/pr-233/a7f7c4dccc69/
- The review shows Apps (7), with War on Disease first and Optimitron second.
- Compatibility checks contain the temporary War on Disease, dFDA, and DIH host captures.
- Native app links use peer previews. Compatibility links use the legacy Optimitron host variant.
- The protocol-v3 baseline refresh is labeled explicitly.
- Stalled font readiness is bounded. War on Disease capture completed in 9m25s after the fix.
- All required checks pass. No unresolved review threads remain.

### PR DESCRIPTION
**Pull request by mikepsinn**
## Summary

- Declare and reconcile one Vercel project per `apps/*` peer application.
- Keep live pull-request previews and production deployments for every affected app.
- Skip unchanged apps with Vercel affected-project detection plus a dependency-aware ignored-build step.
- Disable `gh-pages` deployments so generated visual-review publishing never starts app builds.
- Link each affected app and screenshot route to that app's own non-stale preview.
- Keep Optimitron's custom production path only for pre-deploy database migration and managed-data ordering.

## Cost control

- Current-cycle Vercel usage reached $15.30 of the $20 included credit.
- Build CPU produced $11.77 of the infrastructure usage.
- The six new projects represented about 86% of this app group's build CPU during migration and deployment-config work.
- Workflow-only pushes now cancel all seven ignored builds in about 16-20 seconds instead of performing roughly three-minute builds.
- Live previews remain available for affected apps; this does not trade away pre-merge verification.

## Deployment and review model

- Each app has its own Vercel project, environment variables, deployments, and rollbacks.
- App source, transitive workspace dependencies, and root dependency inputs trigger only affected projects on pull requests and `main`.
- GitHub Actions builds and captures every relevant app for fixture-based visual review.
- The review packet resolves each affected app's latest valid preview and never falls back behind that app's newest change.
- Optimitron's deeper Playwright suite and custom production workflow reflect its current route and database surface, not higher application importance.
- The reconciler does not move production domains or copy secrets.

## Verification

- 70 focused deployment, preview, smoke, and workflow tests pass.
- 18 focused visual readiness and coverage tests pass.
- `git diff --check` passes.
- Live changed-app commits produced previews for all seven apps.
- The required `pr-validate` gate now waits for every affected app preview and the published visual review before merge.
- A CureDAO timeline false positive was traced to an uncommitted React capture state; explicit readiness plus two byte-identical repeat captures verified the fix.
- Workflow-only and visual-review-only commits canceled all seven Vercel builds through the shared ignore script in about 15-20 seconds.
- Non-Vercel Preview environment events no longer start a Playwright smoke job that waits 20 minutes for an impossible Vercel URL.
- Shallow Vercel clones now fetch the exact comparison baseline from GitHub, so a missing local SHA cannot fall through into a full build.
- Browser jobs verify preinstalled Chrome libraries instead of downloading optional system fonts from a slow package mirror on every run.
- Visual capture font waits are bounded at 10 seconds; the War on Disease capture now finishes in 9m25s instead of timing out after 30 minutes.
- Required production database bindings were added to War on Disease, dFDA, Wishocracy, and Trial Abundance Survey; redeploys and route smoke checks passed with no recent runtime errors.
- App preview status, CI, visual review, and review threads are rechecked after every push.

## Summary by CodeRabbit

* **New Features**
  * Preview reviews now support separate URLs and authenticated route links for each affected app.
  * Deployments can build and preview only apps affected by a change.
  * Added a workflow to review and reconcile production app projects.
  * Visual reviews now group routes by app and compatibility checks.
* **Bug Fixes**
  * Improved visual review reliability by waiting for pages to finish preparing.
  * Corrected preview routing and deployment behavior across apps.
* **Documentation**
  * Updated guidance for independent app previews and production deployments.
* **Tests**
  * Expanded coverage for previews, deployment scope, and visual readiness.

## Peer-app visual review

- Closes #239 under parent #238.
- Shows War on Disease, Optimitron, dFDA, Wishocracy, Trial Abundance Survey, CureDAO, and Accelerated Medicine as seven peer apps.
- Moves temporary War on Disease, dFDA, and DIH host captures into Compatibility checks.
- Captures Optimitron explicitly with `site=optimitron` and labels the capture protocol v3 baseline refresh.
- Preserves filtering, deep links, verdict storage, preview links, and mobile behavior.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,6 +45,47 @@ jobs:
...
+        env:
+          BEFORE_SHA: ${{ github.event.before }}
+          CURRENT_SHA: ${{ github.sha }}
+        run: |
+          set -euo pipefail
+          if [[ "$BEFORE_SHA" =~ ^0+$ ]] || ! git cat-file -e "$BEFORE_SHA^{commit}"; then
+            echo "web_deploy=true" >> "$GITHUB_OUTPUT"
+            exit 0
+          fi
+          git diff --name-only "$BEFORE_SHA" "$CURRENT_SHA" > "$RUNNER_TEMP/production-changed-files.txt"
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply .github/workflows/ci.yml:67]: Addressed in 58aa6907: main runs no longer cancel each other, and production deploys queue FIFO.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -43,6 +45,47 @@ jobs:
...
+          node --input-type=module <<'NODE'
+          import { appendFileSync, readFileSync } from "node:fs";
+          import { getVercelPreviewBuildMatches } from "./.github/scripts/preview-smoke-scope.mjs";
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply .github/workflows/ci.yml:70]: Addressed in 58aa6907: the Fix AI corpus now participates in both deployment scope checks, with regression coverage.

**Comment by mikepsinn** [comment]
Addressed both smoke edge cases in e500dc25: unknown projects now reach the existing skip path, and Slack only runs after an actual failed smoke result. Added regression coverage; all 51 workflow tests pass locally.

File: .github/workflows/vercel-app-projects.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+jobs:
+  reconcile:
+    name: reconcile-vercel-app-projects
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    environment: Production
+    env:
+      VERCEL_ORG_ID: ${{ vars.VERCEL_ORG_ID }}
+      VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply .github/workflows/vercel-app-projects.yml:26]: Addressed in 0d1308a6 with a fixed concurrency group and `cancel-in-progress: false`, so manual reconciliation runs queue.

File: apps/acceleratedmedicine/vercel.json [hunk 4]
```diff
@@ -1,6 +1,12 @@
 {
   "$schema": "https://openapi.vercel.sh/vercel.json",
   "framework": "nextjs",
+  "git": {
+    "deploymentEnabled": {
+      "main": true,
+      "*": false
+    }
+  },
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply apps/acceleratedmedicine/vercel.json:9]: Not changing this: Vercel empirically matched `*` against this PR's slash-containing `feature/vercel-app-deployments` ref and canceled all six satellite builds at zero duration. `**` is undocumented here and would add no demonstrated coverage.

[the remark this answers is not in the document]
Comment (mikepsinn) [reply apps/acceleratedmedicine/vercel.json:9]: Correction after checking the build logs: this review was right that `*` did not block the slash-named feature branch. The zero-duration activation deployment was skipped as unaffected, not by that wildcard. The branch-wide preview block has now been removed because affected apps should have live previews; every app instead disables only `gh-pages` and uses a dependency-aware ignore command to skip unrelated global changes.

File: apps/optimitron/scripts/build-visual-review.mjs [hunk 5]
```diff
@@ -1443,6 +1458,7 @@ function buildReviewManifest(groups, coverage) {
     repo: repoSlug,
     previewBaseUrl: pageLinkBaseUrl ? pageLinkBaseUrl.toString() : null,
     reviewUrl: reviewBase ? `${reviewBase}/latest.html` : null,
+    baselineDescription: buildBaselineDescription(),
```
[the remark this answers is not in the document]
Comment (mikepsinn) [reply apps/optimitron/scripts/build-visual-review.mjs:1483]: Valid. Fixed in 3d3ab27a with a per-route baselineSkippedForCaptureProtocol field plus mixed-app and web-only regression coverage.
